src/components/lib/sl/sl.c

```diff
@@ -343,20 +411,36 @@ sl_thd_alloc(cos_thd_fn_t fn, void *data)
 	struct sl_thd *t;
 
 	sl_cs_enter();
-	t = sl_thd_alloc_intern(fn, data, NULL, 0);
+	t = sl_thd_alloc_intern(fn, data);
 	sl_cs_exit();
 
 	return t;
 }
 
-/* Allocate a thread that executes in the specified component */
 struct sl_thd *
-sl_thd_comp_alloc(struct cos_defcompinfo *comp)
+sl_thd_aep_alloc(cos_aepthd_fn_t fn, void *data, int owntc)
 {
 	struct sl_thd *t;
 
 	sl_cs_enter();
-	t = sl_thd_alloc_intern(NULL, NULL, comp, 1);
+	t = sl_thd_aep_alloc_intern(fn, data, NULL, owntc ? SL_FLAG_OWNTC: 0);
+	sl_cs_exit();
+
+	return t;
+}
+
+/* sl object for inithd in the child comp */
+struct sl_thd *
+sl_thd_comp_init(struct cos_defcompinfo *comp, int owntc)
```

Comment:
We probably don't. 
I think there is still a missing piece here, we have `cos_defkernel_api` child alloc api where we use parent's `tc` and also child component doesn't have a scheduler. 
So in that case, we need to make this flag `int is_sched` and turn the flag into `XXX_SEND` or into `0` as it's just an AEP with parent's tcap. 
In that case, I think we'll need invocation API in `sl` library for the child to alloc more threads to be handled by parent scheduler?